diff -durN -x '*~' mISDNuser.orig/Makefile mISDNuser/Makefile
--- mISDNuser.orig/Makefile	2008-09-18 08:14:29.000000000 +0200
+++ mISDNuser/Makefile	2009-06-12 20:21:25.000000000 +0200
@@ -36,7 +36,7 @@
 
 export CFLAGS
 
-SUBDIRS := lib bridge tools example l1oip
+SUBDIRS := lib bridge tools l1oip
 
 
 all:
diff -durN -x '*~' mISDNuser.orig/lib/Makefile mISDNuser/lib/Makefile
--- mISDNuser.orig/lib/Makefile	2008-09-18 08:14:29.000000000 +0200
+++ mISDNuser/lib/Makefile	2009-06-12 20:21:12.000000000 +0200
@@ -2,13 +2,18 @@
 AR = ar
 RANLIB = ranlib
 
-all: libmisdn.a libmisdn_pic.a libmisdn.so
+SOMAJOR=0
+SOMINOR=0.0
+SOVERSION=$(SOMAJOR).$(SOMINOR)
+
+all: libmisdn.a libmisdn_pic.a libmisdn.so.$(SOVERSION)
 
 install:
 	install -d $(INSTALL_PREFIX)$(INSTALL_LIBDIR)
 	install -m 644 libmisdn.a $(INSTALL_PREFIX)$(INSTALL_LIBDIR)
 	install -m 644 libmisdn_pic.a $(INSTALL_PREFIX)$(INSTALL_LIBDIR)
-	install -m 644 libmisdn.so $(INSTALL_PREFIX)$(INSTALL_LIBDIR)
+	install -m 644 libmisdn.so.$(SOVERSION) $(INSTALL_PREFIX)$(INSTALL_LIBDIR)
+	ln -s libmisdn.so.$(SOVERSION) $(INSTALL_PREFIX)$(INSTALL_LIBDIR)/libmisdn.so
 
 MISDN_OBJ = debug.o mbuffer.o q931.o fsm.o mtimer.o mlayer3.o layer3.o dss1user.o dss1net.o
 
@@ -29,8 +34,8 @@
 	$(AR) cru $@ $^
 	$(RANLIB) $@
 
-libmisdn.so: $(MISDN_OBJ)
-	$(CC) $(CFLAGS) -shared -Xlinker -x -o $@ $^
+libmisdn.so.$(SOVERSION): $(MISDN_OBJ)
+	$(CC) $(CFLAGS) -shared -Xlinker -x -Wl,-soname,libmisdn.so.$(SOMAJOR) -o $@ $^
 
 .c.o:
 	$(CC) $(CFLAGS) -o $@ -c $<
